packages/stryker-babel/src/BabelTranspiler.ts

```diff
@@ -0,0 +1,87 @@
+import { Transpiler, TranspilerOptions, TranspileResult, FileLocation } from 'stryker-api/transpile';
+import { File, TextFile, FileKind } from 'stryker-api/core';
+import * as babel from 'babel-core';
+import * as path from 'path';
+import { EOL } from 'os';
+import { CONFIG_KEY_OPTIONS } from './helpers/keys';
+
+class BabelTranspiler implements Transpiler {
+  private babelConfig: babel.GeneratorOptions;
+  private knownExtensions: string[];
+
+  public constructor(options: TranspilerOptions) {
+    this.babelConfig = options.config[CONFIG_KEY_OPTIONS];
+
+    this.knownExtensions = ['.js', '.jsx', '.ts'];
+  }
+
+  public transpile(files: File[]): Promise<TranspileResult> {
+    const outputFiles: File[] = [];
+    const errors: string[] = [];
+
+    files.forEach((file) => {
+      if (this.shouldBeTranspiled(file)) {
+        try {
+          outputFiles.push({
+            name: file.name,
+            content: this.transform((file as TextFile).content),
```

Comment:
You could also use a typeguard for this, but you can leave it as well.

Reply:
We could, but the guard is already implemented (separate method: `shouldBeTranspiled`)